code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/MultiIconNotificationBuilder.kt

```diff
@@ -0,0 +1,131 @@
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewClickAction
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewImage
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MultiIconPushTemplate
+import com.adobe.marketing.mobile.services.Log
+
+internal object MultiIconNotificationBuilder {
+    const val SELF_TAG = "MultiIconNotificationBuilder"
+
+    fun construct(
+        context: Context,
+        pushTemplate: MultiIconPushTemplate,
+        trackerActivityClass: Class<out Activity>?
+    ): NotificationCompat.Builder {
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building an icon template push notification."
+        )
+
+        val packageName = context.packageName
+        val notificationLayout = RemoteViews(packageName, R.layout.push_template_multi_icon)
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channelIdToUse: String = notificationManager.createNotificationChannelIfRequired(
+            context,
+            pushTemplate
+        )
+
+        populateIconsForMultiIconTemplate(
+            context,
+            trackerActivityClass,
+            notificationLayout,
+            pushTemplate,
+            pushTemplate.templateItemList,
+            packageName
+        )
+
+        setCancelIcon(
+            notificationLayout,
+            pushTemplate,
+            packageName)
+
+        notificationLayout.setRemoteViewClickAction(
+            context,
+            trackerActivityClass,
+            R.id.five_icon_close_button,
+            null,
+            null,
+            pushTemplate.tag,
+            false
+        )
+
+        return AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            notificationLayout,
+            notificationLayout,
+            R.id.carousel_container_layout
+        )
+    }
+
+    private fun setCancelIcon(
+        notificationLayout: RemoteViews,
+        pushTemplate: MultiIconPushTemplate,
+        packageName: String?
+    ) {
+        val iconString = pushTemplate.cancelIcon
+        notificationLayout.setRemoteViewImage(iconString, R.id.five_icon_close_button)
+    }
+
+    private fun populateIconsForMultiIconTemplate(
+        context: Context,
+        trackerActivityClass: Class<out Activity>?,
+        notificationLayout: RemoteViews,
+        pushTemplate: MultiIconPushTemplate,
+        items: MutableList<MultiIconPushTemplate.MultiIconTemplateItem>,
+        packageName: String?
+    ) {
+        var validImagesAddedCount = 0
+        for (item in items) {
+            val imageUri: String = item.iconUrl
```

Comment:
nit: just use item.iconUrl directly in line 19 below.

Reply:
Done